### ISSUE TITLE
Tutorial: Source and data for tree example

### ISSUE THREAD

**Issue by sabman** [DataSystemsLab/GeoSpark#9]
Can we have access to the complete example of the tutorial one the website? I am having trouble trying to replicate the example https://gist.github.com/jiayuasu/667c466f313dd4f2fc1b 

Especially mixing spatial and non-spatial features

**Comment by sabman** [comment]
Also I can't see `getSpatialPairRDD()` function anywhere in the source code so it seem the example is completely out of date

**Comment by jiayuasu** [comment]
Hi @sabman , for current version, we separate spatial operator (e.g. range query, join, and KNN) with SpatialRDDs for further tuning performance. 

The usage of one new join function is like this:

```java
SpatialJoinQueryUsingIndex (JavaSparkContext sc, PointRDD pointRDD, RectangleRDD rectangleRDD, String indexType)
```

Two input datasets: one is PointRDD, another is RectangleRDD.
Function returns format (per row): `<a Rectangle, a list of Point>`

[Check out this java class.](https://github.com/DataSystemsLab/GeoSpark/blob/master/src/main/java/org/datasyslab/geospark/spatialOperator/NewJoinQuery.java)

Now our team is fully focusing on GeoSpark. Its developing progress will be accelerated. Detailed tutorials and updated Java API doc will be released very soon. Once it is done, I will close this pull request.

**Comment by sabman** [comment]
@jiayuasu thanks for this! I'll give it a try today.
 I'd really to help with development too, maybe we can connect on google hangout sometime and discuss some of the specs and future direction.

**Comment by sabman** [comment]
@jiayuasu OK I got it working. But it seems to me there is no way to create an RDD with a tuple containing a mix of spatial and non-spatial attributes. Below is an example that is using the csv files in the test directory. I am using PySpark. I'd like to do the join between the RectangleRDD and PointRDD datasets but in the point data also have some non-spatial attributes from the other columns in the csv. Do you have a suggestion for how to do this?

```python
inputLocation = "primaryroads.csv"
offset = int(props['offset'])
gridType = props['gridType']
splitter = props['splitter']
numPartitions = int(props['numPartitions'])
                
rectangleRDD = geospark.spatialRDD.RectangleRDD(sc._jsc, 
    inputLocation, 
    offset, 
    splitter, 
    numPartitions)

pointRDD = geospark.spatialRDD.PointRDD(sc._jsc, 
    inputLocation, 
    offset, 
    splitter, 
    gridType, 
    numPartitions)
pointRDD.buildIndex("R-Tree")

join = jvm.org.datasyslab.geospark\
    .spatialOperator\
    .NewJoinQuery\
    .SpatialJoinQueryUsingIndex(sc._jsc, pointRDD, rectangleRDD, "R-Tree")

print join.first()
# (Env[-122.176411 : -122.174104, 47.576469 : 47.583845],[POINT (-122.176411 47.576469)])

roadsRDD = sc.textFile(inputLocation)
print roadsRDD.take(1)
# [u'0105000000010000000102000000070000009607E92972205CC0D7666325E6FD46404C6E14596B205CC0B7C83D5DDDFD4640255D33F966205CC073ECA012D7FD46403F3BE0BA62205CC0CB4A9352D0FD464082C64CA25E205CC06E46CD57C9FD4640AABA473657205CC087B9A46ABBFD4640A22AA6D24F205CC011A0A696ADFD4640,
#1,-112.506968,45.98186,-112.504872,45.983586,1104469230020,I- 15,I,S1100']

```

**Comment by jiayuasu** [comment]
@sabman Yes. That is an issue. In current implementation, we drop non-spatial attributes during calculation for saving necessary memory overhead. There is no workaround for this version. In the next version, an unique ID will be assigned  to each spatial object after dropping non-spatial attributes.

**Comment by sabman** [comment]
@jiayuasu do you mean a unique ID supplied by the source data? OR automatically generated and assigned by GeoSpark? I personally would like to see the **option** of supplying non-spatial fields. 

I don't really get the idea of dropping attributes all together. How much memory does that really save? I think is a good idea to let the user choose. Anyway I'll try to dig into the code to see how hard it would be to customize this.

**Comment by Sarwat** [comment]
I totally agree with @sabman that the use should have the total freedom to keep or drop non-spatial attributes.

**Comment by jiayuasu** [comment]
Tutorial, Java API doc and GeoSpark website example have been updated. Now, users can get detailed explanation for every public SRDD API in debug branch. The master branch code is stable but some old APIs will be deprecated in the future. This issue is done. The discussion of keeping non-spatial attributes will be moved to a new issue.
